File: src/main/java/net/gtaun/shoebill/constant/ConnectionStatus.java
package net.gtaun.shoebill.constant;

/**
 * Created by marvin on 14.05.15 in project shoebill-api.
 * Copyright (c) 2015 Marvin Haschker. All rights reserved.
 */
public enum ConnectionStatus {
    NO_ACTION,
    DISCONNECT_ASAP,
    DISCONNECT_ASAP_SILENTLY,
    DISCONNECT_ON_NO_ACK,
    REQUESTED_CONNECTION,
    HANDLING_CONNECTION_REQUEST,
    UNVERIFIED_SENDER,
    SET_ENCRYPTION_ON_MULTIPLE_16_BYTE_PACKET,
    CONNECTED
}

File: src/main/java/net/gtaun/shoebill/constant/ConnectionStatus.kt
package net.gtaun.shoebill.constant

/**
 * This enum contains the possible connection statuses that can be used with NetStats_ConnectionStatus.
 *
 * @author Marvin Haschker
 */
enum class ConnectionStatus(val id: Int) {
    NO_ACTION(0),
    DISCONNECT_ASAP(1),
    DISCONNECT_ASAP_SILENTLY(2),
    DISCONNECT_ON_NO_ACK(3),
    REQUESTED_CONNECTION(4),
    HANDLING_CONNECTION_REQUEST(5),
    UNVERIFIED_SENDER(6),
    SET_ENCRYPTION_ON_MULTIPLE_16_BYTE_PACKET(7),
    CONNECTED(8);

    companion object : Collectable<ConnectionStatus>, Findable<Int, ConnectionStatus> {
        private val VALUES = values().map { it.id to it }.toMap()

        /**
         * Gets a [ConnectionStatus] by its id.
         */
        @JvmStatic
        override operator fun get(value: Int): ConnectionStatus? = VALUES[value]

        /**
         * Gets all available [ConnectionStatus]s.
         */
        @JvmStatic
        override fun get(): Collection<ConnectionStatus> = VALUES.values
    }
}
